src/components/stepper/Stepper.tsx

```diff
@@ -0,0 +1,67 @@
+import StepperCircle from "./StepperCircle"
+import { MainContainer } from "./Stepper.styles";
+import { createRef, useRef, useState } from "react";
+import { Button, ButtonsContainer, StepperContainer, FormContainer } from "./StepperHandler.styles";
+
+type Step = {
+    title: string;
+    component: React.ForwardRefExoticComponent<any>;
+    props?: any;
+}
+
+
+interface StepperProps {
+    steps: Step[];
+}
+
+
+function Stepper({ steps }: StepperProps) {
+
+    const [currentStep, setCurrentStep] = useState<number>(0);
+    const formRefs = useRef(steps.map(() => createRef<any>()));
+
+    const nextStep = async () => {
+        const currentFormRef2 = formRefs.current[currentStep];
+        currentFormRef2.current.submit().then((res: any) => {
+            if (res) {
+                setCurrentStep((prev) => Math.min(prev + 1, steps.length - 1));
+            }
+        });
+    };
+
+    const backStep = () => {
+        setCurrentStep((prev) => Math.max(prev - 1, 1)); 
+    };
+
+
+    const maxSteps = steps.length;
+    const CurrentForm = steps[currentStep].component;
+    const currentFormRef = formRefs.current[currentStep];
+
+    if (!steps) {
+        return null;
+    }
```

Comment:
Esto lo deberiamos subir arriba de todo (incluso arriba de donde se se declara el currentStep, además, agregar una validacion para cuando los pasos son un array vacio, porque ahora solo estamos comprobando que los pasos no sean null. Y tambien un mensajito pavo para indicar por que no se muestra ningun paso (basicamente porque no se envio ninguno)

```suggestion
    if (!steps || steps.length === 0) {
        return <div>No steps provided</div>;
    }
```